### PR TITLE
Actually enable fat db, and do RPCs for it.

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: parity/run.rs [hunk 1]
```diff
@@ -115,7 +117,20 @@ pub fn execute(cmd: RunCmd) -> Result<(), String> {
...
+		Colour::White.bold().paint(algorithm.as_str()),
+		match cmd.fat_db {
+			Switch::On if algorithm == Algorithm::Archive => format!("{}", Colour::White.bold().paint(" +Fat")), 
+			Switch::On => die("Fat DB is not supported with the chosen pruning option. Please rerun with `--pruning=archive`"),
+			Switch::Auto => die("--fat-db auto is not supported."),
+			Switch::Off => "".to_owned(),
+		},
+		match cmd.tracing {
+			Switch::On => format!("{}", Colour::White.bold().paint(" +Trace")),
+			Switch::Auto => format!("{}", Colour::White.bold().paint(" +?Trace")),
```
Comment (debris) [inline parity/run.rs]: it can be replaced with shorter: `Colour::White.bold().paint(" +?Trace").to_string()`

File: parity/run.rs [hunk 2]
```diff
@@ -337,3 +353,9 @@ fn wait_for_exit(
 	let _ = exit.wait(mutex.lock().unwrap());
 	info!("Finishing work, please wait...");
 }
+
+
+fn die(msg: &'static str) -> ! {
+	println!("{}", msg);
+	::std::process::exit(-1)
```
Comment (debris) [inline parity/run.rs]: I removed `die`, cause it breaks tests execution

File: ethcore/src/client/client.rs [hunk 3]
```diff
@@ -169,6 +170,11 @@ impl Client {
 		let db = Arc::new(try!(Database::open(&db_config, &path.to_str().unwrap()).map_err(ClientError::Database)));
 		let chain = Arc::new(BlockChain::new(config.blockchain, &gb, db.clone()));
 		let tracedb = Arc::new(try!(TraceDB::new(config.tracing, db.clone(), chain.clone())));
+		let trie_spec = match config.fat_db {
+			Switch::On => TrieSpec::Fat,
+			Switch::Off => TrieSpec::Secure,
+			Switch::Auto => unimplemented!(),
```
Comment (debris) [inline ethcore/src/client/client.rs]: it should be implemented before merge

**Comment by gavofyork** [comment]
onice since #2048 should include this functionality

**Pull request closed without merging**